### PR TITLE
Amanjeev/tyk authn no dashboard

### PR DESCRIPTION
**Pull request by poquirion**

### CODE REVIEW

File: template/compose.tpl/containers_network.yml.tpl [hunk 1]
```diff
@@ -52,7 +42,7 @@ services:
     networks:
     - tyk
   candig_server:
-    image: c3genomics/candig_server
+    build: ../../candig-server
```
Comment (poquirion) [inline template/compose.tpl/containers_network.yml.tpl]: I think we should not publish a hardcode the build step ;)

File: template/config.tpl/tyk/confs/api_candig.json.tpl [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    "upstream_certificates": {},
+    "use_keyless": false,
+    "enable_coprocess_auth": false,
+    "base_identity_provided_by": "",
+    
+    "proxy": {
+	"target_url": "${LOCAL_CANDIG_SERVER}",
+	"strip_listen_path": true,
+        "disable_strip_slash": false,
+	"listen_path": "/${TYK_LISTEN_PATH}",
```
Comment (poquirion) [inline template/config.tpl/tyk/confs/api_candig.json.tpl:21]: Not sure about the `/`. Is it supposed to be there?

File: template/config.tpl/tyk/confs/api_candig.json.tpl [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	],
+	"TYK_SERVER": "${CANDIG_PUBLIC_URL}${CD_PUB_PORT}"
+    },
+    "openid_options": {
+	"segregate_by_client": false,
+	"providers": [
+            {
+                "issuer": "${KC_PUBLIC_URL}${KC_PUB_PORT}/auth/realms/${KC_REALM}",
+                "client_ids": {
+                    "Y3FfY2FuZGln": "policy21"
```
Comment (poquirion) [inline template/config.tpl/tyk/confs/api_candig.json.tpl]: That is the base64 of something? Is it always the same and really need to be hardcoded?

**Review by poquirion — COMMENTED** [review]
I am a bit surprised about all the hard coded id and api names. We should make sure it is the right way of doing it.

File: template/compose.tpl/containers_network.yml.tpl (hunk 1 above)
Comment (amanjeev) [reply template/compose.tpl/containers_network.yml.tpl]: Yea sorry totally forget this. Will remove.

File: template/config.tpl/tyk/confs/api_candig.json.tpl (hunk 2 above)
Comment (amanjeev) [reply template/config.tpl/tyk/confs/api_candig.json.tpl:21]: I think it was there before?

File: template/config.tpl/tyk/confs/api_candig.json.tpl (hunk 3 above)
Comment (amanjeev) [reply template/config.tpl/tyk/confs/api_candig.json.tpl]: So I am afraid that I let this slip in. It is your `cq_candig` :)

Comment (amanjeev) [reply template/config.tpl/tyk/confs/api_candig.json.tpl]: I should perhaps make it dynamic.

File: template/config.tpl/tyk/confs/api_candig.json.tpl (hunk 2 above)
Comment (amanjeev) [reply template/config.tpl/tyk/confs/api_candig.json.tpl:21]:
```
# listen_path is empty "", put a slash only if there is a path "/<path>"
export TYK_LISTEN_PATH=""
```

I think we do not need that slash?

**Comment by amanjeev** [comment]
> I am a bit surprised about all the hard coded id and api names. We should make sure it is the right way of doing it.

Truth be told, I really did not have time to go and dig further to make them dynamic. I was hoping for iterative improvements once I get this rolling. @SamarthVP is perhaps a good candidate to take on the task of making these things dynamic. For now, we really do not need more than one policy and more than one or two APIs. That is what I feel.

I have spent most of the day today working on adding a new API behind Tyk and documenting it. So that should help with some of this stuff.

**Comment by amanjeev** [comment]
Okie doke! :) I made the changes and templatized to as much as I could.

**Comment by poquirion** [comment]
Ok, I am ready for the merge!

**Comment by amanjeev** [comment]
@ljdursi Somehow missed your comment. Perhaps a new PR is needed. 

> two possibilities are that if a key isn't explicitly provided to (a) use a randomly-generated key, or (b) halt with an error?

Who should be working on that?

**Pull request merged**